src/main/java/io/anserini/collection/WashingtonPostCollection.java

```diff
@@ -53,4 +58,137 @@ public FileSegment(Path path) throws IOException {
   public FileSegment createFileSegment(Path p) throws IOException {
     return new WashingtonPostCollection.FileSegment(p);
   }
+
+  /**
+   * A document from the Washington Post collection.
+   */
+  public static class Document implements SourceDocument {
+    private static final Logger LOG = LogManager.getLogger(Document.class);
+    private static final String PATTERN = "<\\/?\\w+>";
+    private final List<String> CONTENT_TYPE_TAG = Arrays.asList("sanitized_html", "tweet");
+
+    // Required fields
+    protected String id;
+    protected long published_date;
+    protected String content;
+
+    @Override
+    public SourceDocument readNextRecord(BufferedReader bufferedReader) throws IOException {
+      String nextRecord = bufferedReader.readLine();
+      if (nextRecord == null) {
+        return null;
+      }
+      return parseRecord(nextRecord);
+    }
+
+    private SourceDocument parseRecord(String record) {
+      StringBuilder builder = new StringBuilder();
+      ObjectMapper mapper = new ObjectMapper();
+      WashingtonPostObject recordObj = null;
+      try {
+        recordObj = mapper
+                .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES) // Ignore unrecognized properties
+                .registerModule(new Jdk8Module()) // Deserialize Java 8 Optional: http://www.baeldung.com/jackson-optional
+                .readValue(record, WashingtonPostObject.class);
+      } catch (IOException e) {
+        // For current dataset, we can make sure all record has unique id and
+        //  published date. So we just simply log a warning and return null
+        //  here in case future data may bring up this issue
+        LOG.warn("No unique ID or published date for this record, ignored...");
+        return null;
+      }
+
+      id = recordObj.id();
+      published_date = recordObj.published_date();
+
+      if (recordObj.contents != null && recordObj.contents.isPresent()) {
```

Comment:
This block seems to be to verbose, can we make it simpler?

Reply:
When a field such as `contents` is missing in the JSON string, `recordObj.contents` will be equal to `null` after deserialization. When the JSON string contains `{"contents" : null}`, `recordObj.contents` will be equal to `Optional.empty()` after deserialization.

That's why I've used `if (recordObj.contents != null && recordObj.contents.isPresent())`.

Reply:
That makes sense. I'll do it now.